### PR TITLE
[WIP] Set the permissions on the upper diff directory based on the lower dir

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
We have a bug where a user sets the Image / to chmod 555, but when you run a container based on Overlay on top, it ends up running as 755.  This is an attempt to fix.

**Comment by TomSweeneyRedHat** [comment]
Looks like some tests will need adjusting, Travis isn't happy.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Tests finally pass, now I got to merge this into buildah to see if it actually works.

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -441,6 +442,15 @@ func (d *Driver) create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		return err
 	}
 	if lower != "" {
+		lowers := strings.Split(lower, ":")
+		newpath := path.Join(d.home, lowers[0])
```
Comment (nalind) [inline drivers/overlay/overlay.go]: This will fail if the parent layer is in a read-only store.  Try `newpath := d.dir(lowers[0])` instead.

**Comment by rhatdan** [comment]
Have we basically decided on this that this needs to be fixed somewhere else?

**Comment by nalind** [comment]
Layer blobs omit the root directory, so I don't think this can fix all of the cases.

**Comment by rhatdan** [comment]
Closing since this is not an easily solved problem

**Pull request closed without merging**